### ISSUE TITLE
Inconsistent behavior between setSize and setScale regarding origin

### ISSUE THREAD

**Issue by kaltinril** [libgdx/libgdx#3272]
I've been having some issues trying to get an image to rotate around a point and I've found part of my issue is that inconsistent behavior of setSize and setScale.

If the origin is set, then the size, then rotated, then drawn, the image is pivoted at the origin, which is not adjusted or corrected for after a resize.

If the sprite is instead scaled, the origin does not affect it.

To me calling setSize should somehow "adjust" the origin based on where it previously was.

Also, side note, what is the difference between setSize and setScale to a user of the sprite class?

Essentially to get setSize to work correctly the order of calling methods must be:
```Java
sprite.setSize(w,h);
sprite.setOriginCenter(); //This must be called AFTER a resize when using setSize.
sprite.setRotation(180);
```

See below full code example using assets from the [gdx-tests-android](https://github.com/libgdx/libgdx/tree/master/tests/gdx-tests-android/assets/data) project:

```Java

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.backends.lwjgl.LwjglApplication;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.graphics.Texture;

public class DesktopLauncher extends ApplicationAdapter{
	public static void main (String[] arg) throws Exception {
		new LwjglApplication(new DesktopLauncher());
	}

	SpriteBatch batch;
	Texture img;

	//Setup some sprite tests
	Sprite spriteNoRotation;
	Sprite spriteCenterThenSetSizeThenRotate;
	Sprite spriteSetSizeThenCenterThenRotate;

	@Override
	public void create () {
		batch = new SpriteBatch();
		img = new Texture(Gdx.files.internal("data/black_marked_0.png"));

		//Resize the image and move it up
		spriteNoRotation = new Sprite(img);
		spriteNoRotation.setOrigin(5, 5);
		spriteNoRotation.setSize(img.getWidth() / 2, img.getHeight() / 2);
		spriteNoRotation.setCenter(img.getWidth(), img.getHeight() + (img.getHeight() / 2) + (img.getHeight() / 4)); // setCenter puts the image at the location, centered at the middle of the image (Width/height)

		//Resize the image and move it up
		//Then this rotates around the origin BEFORE the size was adjusted
		//This causes the image to be rotated at the top right of the image, instead of the center.
		spriteCenterThenSetSizeThenRotate = new Sprite(img);
		spriteCenterThenSetSizeThenRotate.setOriginCenter();          //This is the key, setOriginCenter() is run first, when the image is 64 by 64
		spriteCenterThenSetSizeThenRotate.setSize(img.getWidth() / 2, img.getHeight() / 2);
		spriteCenterThenSetSizeThenRotate.setCenter(img.getWidth(), img.getHeight() + (img.getHeight() / 2) + (img.getHeight() / 4));
		spriteCenterThenSetSizeThenRotate.setRotation(180);           //This should just be turned upside down, but instead this rotates around the top right point

		spriteSetSizeThenCenterThenRotate = new Sprite(img);
		spriteSetSizeThenCenterThenRotate.setSize(img.getWidth() / 2, img.getHeight() / 2);              //Move this so that it happens BEFORE we set the center
		spriteSetSizeThenCenterThenRotate.setOriginCenter();
		spriteSetSizeThenCenterThenRotate.setCenter(img.getWidth(), img.getHeight() + (img.getHeight() / 2) + (img.getHeight() / 4));
		spriteSetSizeThenCenterThenRotate.setRotation(45);
	}

	@Override
	public void render() {
		Gdx.gl.glClearColor(1, 0.5f, 0.5f, 1);
		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
		batch.begin();

		spriteNoRotation.draw(batch);
		spriteCenterThenSetSizeThenRotate.draw(batch);	//Does not work, image rotation point is left at 32x32, which is now the top right of the image, so it pivots
		spriteSetSizeThenCenterThenRotate.draw(batch);	//Works correctly

		batch.end();
	}
}
```

**Comment by xoppa** [comment]
So, if i understand correctly, the issue you're reporting is that: the origin is not adjusted when changing the size. Is that correct? If so, then that's correct. `setSize` sets the size and nothing else. `setOrigin` sets the origin and nothing else. Btw, did you actually search for previous issues (either open or closed)? Because I'm pretty sure this has been reported before.

**Comment by NathanSweet** [comment]
Origin and size are separate properties and should stay that way. I think this PR can be closed.

FWIW, using scale means a group needs to have transform enabled, which causes a batch flush.

**Comment by kaltinril** [comment]
@xoppa , Thanks, I guess in my mind setSize shouldn't allow the origin to get screwy.  I did try searching but its hard to guess what others may have thought their issue was, and how they may have worded their issue/question.

@NathanSweet , Alright, I guess its just my thought that not adjusting the origin after a setSize causes the origin to now be inaccurate, and potentially out of the area of the visible sprite/image.

I guess could I suggest at least adding some sort of comment in any examples or on any wiki pages, that setSize will not adjust the origin?  It may be obvious to those that understand it, and wrote it, but for those who are using it for the first time it may not be.

I think my confusion stems from using XNA and how they defined and interacted with sprites.  Most positioning was done via the center of the image with XNA if you set the origin.

**Final answer:**
If that is working as designed then I agree, close this issue.

Also, I fixed all my rotation issues from my misunderstandings on how the sprite class methods worked.
